src/deepsparse/transformers/helpers.py

```diff
@@ -193,3 +199,93 @@ def _normalize_fields(_dict):
         return result
 
     return _wrapper
+
+
+def truncate_transformer_onnx_model(
+    model_path: str,
+    emb_extraction_layer: int = -1,
+    model_size: Optional[int] = None,
+    final_node_name: Optional[str] = None,
+    output_name: str = "embedding",
+    output_path: Optional[str] = None,
+) -> str:
+    """
+    Determines where to cut the transformer model using best-guess heuristics
+    Saves cut model to output_path or temporary file
+
+    :param model_path: path of onnx file to be cut
+    :param emb_extraction_layer: last bert layer to include.
+        default -1 (last layer)
+    :param final_node_name: name of last computed node in graph
+    :param output_name: name of graph output, default "embedding"
+    :param output_path: path to write resulting onnx file. If not provided,
+        will create a temporary file path that will be destroyed on program end
+    :return: if no output path, a tuple of the saved path to the model, list of
+        model output names, and reference to the tempfile object will be returned
+        otherwise, only the model output names will be returned
+    """
+
+    def _check_initializer_names(model: ModelProto) -> Union[str, None]:
+        bert_layer_prog = re.compile(
+            r"bert\.encoder\.layer\.\d+\.output\.LayerNorm\.bias"
+        )
+        distillbert_layer_prog = re.compile(
+            r"distilbert.transformer.layer.\d+\.output_layer_norm.bias"
+        )
+
+        layer_init_names = [
+            initializer.name
+            for initializer in model.graph.initializer
+            if bert_layer_prog.match(initializer.name)
+            or distillbert_layer_prog.match(initializer.name)
+        ]
+        assert len(layer_init_names) > 0
```

Comment:
what is the purpose of this code block?

Reply:
This code probably needs to be rewritten for clarity. It finds initializers with names which match the above regexes and stores them in `layer_init_names`. This list of initializer names it then used as a proxy for the last nodes in each bert layer.

Reply:
named to `bert_layer_pattern`